src/Microsoft.FeatureManagement/FeatureFilters/Recurrence/RecurrenceValidator.cs

```diff
@@ -413,7 +402,7 @@ private static int CalculateWeeklyDayOffset(DayOfWeek day1, DayOfWeek day2)
         /// </summary>
         private static List<DayOfWeek> SortDaysOfWeek(IEnumerable<DayOfWeek> daysOfWeek, DayOfWeek firstDayOfWeek)
         {
-            List<DayOfWeek> result = daysOfWeek.ToList();
+            List<DayOfWeek> result = daysOfWeek.Distinct().ToList(); // dedup
```

Comment:
Could we take a Set instead so we don't need to Distinct it?

Reply:
I personally like the `Distinct` one, it is more readable. If we use a set here, we will either need to write two lines of code (1 line for new set and 1 line for ToList) or we will write `List<DayOfWeek> result = new HashSet<DayOfWeek>(daysOfWeek).ToList();` I feel this single line ugly. I believe the background implementation of `Distinct` is based on hash set. Do you have any other concern about using `Distinct`